backend/app.py

```diff
@@ -13,29 +17,34 @@ def getinfo():
 
 
 @app.route('/getPastData', methods=['POST'])
-def getPastData():
-    filtering_factor = request.get_json()['filtering_factor']
-    data = read_data.create_formatted_data(filtering_factor, True)
-    return jsonify(data)
+def get_past_data():
+    filter_gender = request.get_json()['filtering_factor']
+
+    past_data = (DataFormatter(women_bias_data)
+                 .filter_by(filter_gender)
+                 .filter_invalid_transactions()
+                 .get_for_display())
+
```

Comment:
This is gonna fail. Your get_for_display returns a pd.DataFrame, which is suitable for model imput, but not here. Here you need to transfer the DataFrame into a list of dictionaries, where each dictionary has two key-value pairs: {'date': a STRING of the date; 'value': number_of_transaction or value_of_transaction}. This conversion is made in the old create_formatted_data in read_data.py.

I recommend you to keep your get_for_display but rename it to get_for_model. Add a new get_for_display to convert the format to list of dicts.